packages/drivers/odsp-driver/src/odspUtils.ts

```diff
@@ -272,8 +272,9 @@ export function toInstrumentedOdspTokenFetcher(
                 // This event alone generates so many events that is materially impacts cost of telemetry
                 // Thus do not report end event when it comes back quickly.
                 // Note that most of the hosts do not report if result is comming from cache or not,
-                // so we can't rely on that here
-                if (event.duration >= 32) {
+                // so we can't rely on that here. But always record if specified explicitly for cases such as
+                // calling trees/latest during load.
+                if (event.duration >= 32 || alwaysRecordTokenFetchTelemetry) {
```

Comment:
Super nit, I would reverse the order such that it's "if want to always record telemetry *or* duration is >= 32"